api/topic_test.go

```diff
@@ -90,92 +102,186 @@ func dbTopicWithId(state models.State, id string) *models.TopicUpdate {
 	}
 }
 
-// API model corresponding to dbCreatedTopic !!! ?
-func createdTopic() *models.TopicUpdate {
+// API model corresponding to dbCreatedTopic
+func createdTopicAll() *models.TopicUpdate {
 	return dbTopic(models.StateTopicCreated)
 }
 
-func TestGetTopicHandler(t *testing.T) {
+// create just the 'current' sub-document
+func dbTopicCurrent(state models.State) *models.Topic {
+	return dbTopicCurrentWithID(state, testTopicID1)
+}
 
-	Convey("Given a topic API in publishing mode", t, func() {
-		cfg, err := config.Get()
-		cfg.EnablePrivateEndpoints = true
-		So(err, ShouldBeNil)
-		doTestGetTopicHandler(cfg)
-	})
+func dbTopicCurrentWithID(state models.State, id string) *models.Topic {
+	return &models.Topic{
+		ID:          id,
+		Description: "test description - 1",
+		Title:       "test title - 1",
+		Keywords:    []string{"keyword 1", "keyword 2", "keyword 3"},
+		State:       state.String(),
+		Links: &models.TopicLinks{
+			Self: &models.LinkObject{
+				HRef: fmt.Sprintf("http://example.com/topics/%s", id),
+				ID:   fmt.Sprintf("%s", id),
+			},
+			Subtopics: &models.LinkObject{
+				HRef: fmt.Sprintf("http://example.com/topics/%s/subtopics", id),
+			},
+			Content: &models.LinkObject{
+				HRef: fmt.Sprintf("http://example.com/topics/%s/content", id),
+			},
+		},
+	}
+}
+
+func createdTopicCurrent() *models.Topic {
+	return dbTopicCurrent(models.StateTopicCreated)
+}
 
-	Convey("Given a topic API in web mode", t, func() {
+func TestGetTopicPublicHandler(t *testing.T) {
+
+	Convey("Given a topic API in web mode (private endpoints disabled)", t, func() {
 		cfg, err := config.Get()
 		cfg.EnablePrivateEndpoints = false
 		So(err, ShouldBeNil)
-		doTestGetTopicHandler(cfg)
-	})
-}
+		Convey("And a topic API with mongoDB returning 'created' and 'current' topics", func() {
```

Comment:
`So(err, ShouldBeNil)`

No harm here but it's good practice to get into the habit of always doing your error checking immediately after the statement that can return an error. It's so easy to forget to do this and can cause much pain.

Reply:
Oh, ur so fussy.
And agreed ... done.